### DISCUSSION TITLE
Exercise 4: Label with proper logical clock

### DISCUSSION THREAD

**Discussion by dag7dev** [sapienzastudentsnetwork/distributed-systems#29] [Exercises]
**Exercise 4:** Label with proper logical clock all the events of the distributed computation in image vector.pdf. (You can consider events that receive a message and immediately send it as single events.)

![image](https://github.com/sapienzastudentsnetwork/distributed-system-2023-2024/assets/44711271/3e3026fc-d5b7-499d-8590-824745395f89)

**Comment by riccardoscuto** [comment]
![image](https://github.com/sapienzastudentsnetwork/distributed-system-2023-2024/assets/11277638/cfdac06e-5957-43f2-aaa5-1f0a4b8e1654)

**Comment by susannacifani** [comment]
Lamport clock:
![image](https://github.com/sapienzastudentsnetwork/distributed-system-2023-2024/assets/73530772/2be0fda4-2d47-424c-aaeb-e8301c5d97cc)

Vector clock:
![image](https://github.com/sapienzastudentsnetwork/distributed-system-2023-2024/assets/73530772/76f4bab3-5ceb-4c0f-8140-79d1aacf0a6e)
